src/plugins/data_source/server/data_source_service.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Logger, OpenSearchClient, SavedObjectsClientContract } from 'src/core/server';
+import { DataSourceClient } from './client';
+import { IDataSourceService } from './types';
+
+export class DataSourceService implements IDataSourceService {
+  private openSearchClientsPool: Map<string, OpenSearchClient>;
+  constructor() {
+    this.openSearchClientsPool = new Map<string, OpenSearchClient>();
+  }
+  // TODO: placeholders, need implement when adding global config
+  isEnabled(): boolean {
+    throw new Error('Method not implemented.');
+  }
```

Comment:
as discussed, let's use datasource plugin enablement as the global config

Reply:
yes, this will be aligned with global config.

Reply:
we are using `data_source.enabled` currently